plugin/trino-opensearch/src/main/java/io/trino/plugin/opensearch/ScanQueryPageSource.java

```diff
@@ -158,8 +158,23 @@ public Page getNextPage()
             Map<String, Object> document = hit.getSourceAsMap();
 
             for (int i = 0; i < decoders.size(); i++) {
-                String field = columns.get(i).name();
-                decoders.get(i).decode(hit, () -> getField(document, field), columnBuilders[i]);
+                OpenSearchColumnHandle columnHandle = columns.get(i);
+                if (!columnHandle.dereferencePath().isEmpty()) {
```

Comment:
Nit: reverse this condition and move the `else` block here with a `continue`. Then you won't need an `else` block at all.